schema/schema.go

```diff
@@ -0,0 +1,50 @@
+package schema
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	// fs stores the embedded http.FileSystem
+	// having the OCI JSON schema files in root "/".
+	fs = _escFS(false)
+
+	specs = map[jsonManifest]string{
+		MediaTypeManifest:     "image-manifest-schema.json",
+		MediaTypeManifestList: "manifest-list-schema.json",
+	}
+)
+
+// FindValidator returns the validator for the given media type
+// or an error if none could be found.
+func FindValidator(mediaType string) (Validator, error) {
```

Comment:
I agree this function is awkward, but don't we already have this through casting? Something like `Validator(mediaType).Validate(r)` is workable.

For reference, this pattern is very close to [crypto.Hash](https://golang.org/pkg/crypto/#Hash). It may help to look there for inspiration.